server/protocols/jmap/src/main/java/org/apache/james/jmap/model/Number.java

```diff
@@ -0,0 +1,134 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.model;
+
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import com.google.common.math.LongMath;
+import com.google.common.primitives.Ints;
+
+public class Number {
+
+    public static final Logger LOGGER = LoggerFactory.getLogger(Number.class);
+
+    public interface Factory {
+        Number fromInt(int value);
+        Number fromLong(long value);
+    }
+
+    public static class StrictFactory implements Factory {
+        public Number fromInt(int value) {
+            Preconditions.checkState(value >= ZERO_VALUE,
+                "value should be positive and less than 2^31 or empty");
```

Comment:
or empty?

Reply:
This error message should display to the end-user so it makes sense to tell them that value should be restrict or let it be empty. WDYT?